### PR TITLE
[release/2.1] Skip CoreFx.Private.TestUtilities in source-build

### PR DESCRIPTION
**Pull request by dagood**
This project referenced the XUnit.Runtime dependency project, causing test-only prebuilt dependencies.

Even though source-build always passes BuildTests=false, include a condition to handle "true" to make intent clear.

---

This is a patch for removing test dependencies from source-build: https://github.com/dotnet/source-build/pull/847.

From what I can tell, `ProjectExclusions` is a vestigial itemgroup from the `dev/eng` branch. I don't see any usages, but we can start using it for this.

I've got a port of this on `master` that I submitted as https://github.com/dotnet/corefx/pull/33218, since `dir.props` => `Directory.Build.props`.

(If this doesn't seem like the right way to do this, let me know a better fix and I can update the patch.)

### CODE REVIEW

File: dir.props [hunk 1]
```diff
@@ -306,6 +306,14 @@
     <SymbolPackageOutputPath Condition="'$(SymbolPackageOutputPath)'==''">$(PackageOutputPath)symbols/</SymbolPackageOutputPath>
   </PropertyGroup>
 
+  <!--
+    Remove test-only dependencies during source-build: disable project that brings in XUnit.Runtime
+    dependencies through 'ReferenceFromRuntime' property.
+  -->
+  <ItemGroup Condition="'$(DotNetBuildFromSource)' == 'true' and '$(BuildTests)' != 'true'">
+    <ProjectExclusions Include="$(SourceDir)CoreFx.Private.TestUtilities/**/*.csproj" />
+  </ItemGroup>
```
Comment (ViktorHofer) [inline dir.props]: Shouldn't that go into `src\Directory.Build.props`?

cc @weshaggard

Comment (dagood) [reply dir.props]: I suppose `src/dir.props` would also work, if that's preferred. (Nothing in there now, but it does exist and it's in the chain.)

Comment (ViktorHofer) [reply dir.props]: sure, I forgot that in 2.1 we don't have the Directory.Build.props changes.

**Comment by dagood** [comment]
Yes, this would let us remove a source-build patch on `2.1`/`2.2`. Closing/opening to refresh CI, @bartonjs can you merge when it makes sense to, and make sure it gets (auto) ported to 2.2? (Not sure if there's a merge procedure on the branch currently.)

**Comment by stephentoub** [comment]
@dagood, what's the status of this?

**Comment by dagood** [comment]
It's waiting to get clean CI without infra issues and be merged. Maybe @dseefeld or @crummel can help push it further.

**Comment by dseefeld** [comment]
@dotnet-bot test Windows x86 Release Build please

**Comment by dseefeld** [comment]
@dotnet-bot test Windows x64 Debug Build please

**Comment by crummel** [comment]
Can't get to the actual results for some reason - another infra failure?
@dotnet-bot test Windows x86 Release Build please
@dotnet-bot test Windows x64 Debug Build please

**Comment by stephentoub** [comment]
@dotnet-bot test Windows x64 Debug Build please
@dotnet-bot test Windows x86 Release Build please

**Comment by dagood** [comment]
This was merged already in https://github.com/dotnet/corefx/pull/35039. Thanks, though. 😄

**Pull request closed without merging**